unity/Assets/Scripts/Unity/Input/MouseEventsHandler.cs

```diff
@@ -0,0 +1,102 @@
+﻿using UnityEngine;
+using UnityEngine.InputSystem;
+using Hexxle.Unity.Util;
+using Hexxle.CoordinateSystem;
+using Hexxle.Unity.Audio;
+
+namespace Hexxle.Unity.Input
+{
+    public class MouseEventsHandler : MonoBehaviour
+    {
+        GameObject oldTile;
+        GameObject currentCollisionTile;
+        public bool isGamePaused { get; private set; } = false;
+
+        private void Awake()
+        {
+            oldTile = null;
+            currentCollisionTile = null;
+        }
+
+        // Start is called before the first frame update
+        void Start()
+        {
+            var inputManager = new InputManager();
+            inputManager.TilePlacement.MouseClick.Enable();
+            inputManager.TilePlacement.MouseClick.performed += context => MouseClickAction();
```

Comment:
I'd move this to OnEnable() and also make sure, that OnDisable() is implemented and strips `performed` of its Eventhandler method. See EscapeEventHandler class.